tests/functional/directives.ts

```diff
@@ -0,0 +1,655 @@
+        const mutation = `mutation {
+          mutationWithUpperAndAppend(append: ", world!")
+        }`;
+
+        const { data } = await graphql(schema, mutation);
+
+        expect(data).toHaveProperty("mutationWithUpperAndAppend");
+        expect(data.mutationWithUpperAndAppend).toBe("HELLO, WORLD!");
+      });
+
+      it("calls directive 'upper' and 'append' using SDL", async () => {
+        const mutation = `mutation {
+          mutationWithUpperAndAppendSDL(append: ", world!")
+      }`;
+
+        const { data } = await graphql(schema, mutation);
+
+        expect(data).toHaveProperty("mutationWithUpperAndAppendSDL");
+        expect(data.mutationWithUpperAndAppendSDL).toBe("HELLO, WORLD!");
+      });
+    });
+
+    describe("InputType", () => {
+      it("adds field directive to input types", async () => {
+        const inputType = schema.getType("DirectiveOnClassInput") as GraphQLInputObjectType;
+
+        expect(inputType).toHaveProperty("astNode");
+        assertValidFieldDirective(inputType.astNode, "upper");
+      });
+
+      it("adds field directives to input type fields", async () => {
+        const fields = (schema.getType(
+          "DirectiveOnFieldInput",
+        ) as GraphQLInputObjectType).getFields();
+
+        expect(fields).toHaveProperty("append");
+        expect(fields.append).toHaveProperty("astNode");
+        assertValidFieldDirective(fields.append.astNode, "upper");
+      });
+    });
+
+    describe("ObjectType", () => {
+      it("calls object type directives", async () => {
+        const query = `query {
+          objectType {
+            withDirective
+            withDirectiveWithArgs
+            withUpper
+            withUpperSDL
+            withAppend(append: ", world!")
+            withAppendSDL(append: ", world!")
+            withUpperAndAppend(append: ", world!")
+            withUpperAndAppendSDL(append: ", world!")
+            withInput(input: { append: ", world!" })
+            withInputUpper(input: { append: ", world!" })
+            withInputOnClass(input: { append: ", world!" })
+            withInputUpperOnClass(input: { append: ", world!" })
+          }
+      }`;
+
+        const { data } = await graphql(schema, query);
+
+        expect(data).toHaveProperty("objectType");
+        expect(data.objectType).toEqual({
+          withDirective: "withDirective",
+          withDirectiveWithArgs: "withDirectiveWithArgs",
+          withUpper: "WITHUPPER",
+          withUpperSDL: "WITHUPPERSDL",
+          withAppend: "hello, world!",
+          withAppendSDL: "hello, world!",
+          withUpperAndAppend: "HELLO, WORLD!",
+          withUpperAndAppendSDL: "HELLO, WORLD!",
+          withInput: "hello, WORLD!",
+          withInputUpper: "HELLO, WORLD!",
+          withInputOnClass: "hello, WORLD!",
+          withInputUpperOnClass: "HELLO, WORLD!",
+        });
+      });
+    });
+  });
+
+  describe("@cacheControl", () => {
```

Comment:
Maybe it should be in separate test file? I'm not a fan of all-in-one test suites

Reply:
@19majkel94 done